# 使用**Anaconda**

[Anaconda](http://www.continuum.io/downloads)是一个很好用的Python IDE，它集成了很多科学计算需要使用的**python**第三方工具包。

## conda 的使用 

根据自己的操作系统安装好[Anaconda](http://www.continuum.io/downloads)后，在命令行下输入：

    conda list

可以看已经安装好的**python**第三方工具包，这里我们使用 `magic` 命令 `%%cmd` 在 `ipython cell` 中来执行这个命令：

In [1]:
%%cmd 
conda list

Microsoft Windows [�汾 6.1.7601]
��Ȩ���� (c) 2009 Microsoft Corporation����������Ȩ����

C:\Users\lijin>conda list
# packages in environment at C:\Anaconda:
#
_license                  1.1                      py27_0  
anaconda                  2.2.0                np19py27_0  
argcomplete               0.8.4                    py27_0  
astropy                   1.0.1                np19py27_0  
atom                      0.3.9                    py27_0  
bcolz                     0.8.1                np19py27_0  
beautiful-soup            4.3.2                    py27_1  
binstar                   0.10.1                   py27_3  
biopython                 1.61                 np17py27_0  
bitarray                  0.8.1                    py27_1  
blaze                     0.6.3                np19py27_0  
blaze-core                0.7.3                np19py27_0  
blz                       0.6.2                np19py27_0  
bokeh                     0.8.1                np19py27_1  
bot

第一次安装好[Anaconda](http://www.continuum.io/downloads)以后，可以在命令行输入以下命令使[Anaconda](http://www.continuum.io/downloads)保持最新：

    conda update conda
    conda update anaconda

conda 是一种很强大的工具，具体用法可以参照它的[文档](http://conda.pydata.org/docs/)。

也可以参考它的[cheat sheet](http://conda.pydata.org/docs/_downloads/conda-cheatsheet.pdf)来快速查看它的用法。

可以使用它来安装，更新，卸载第三方的**python**工具包：

    conda install <some package>
    conda update <some package>
    conda remove <some package>

在安装或更新时可以指定安装的版本号，例如需要使用`numpy 1.8.1`：

    conda install numpy=1.8.1
    conda update numpy=1.8.1

查看 `conda` 的信息：

    conda info

In [4]:
%%cmd
conda info

Microsoft Windows [�汾 6.1.7601]
��Ȩ���� (c) 2009 Microsoft Corporation����������Ȩ����

C:\Users\lijin>conda info
Current conda install:

             platform : win-64
        conda version : 3.14.1
  conda-build version : not installed
       python version : 2.7.5.final.0
     requests version : 2.7.0
     root environment : C:\Anaconda  (writable)
  default environment : C:\Anaconda
     envs directories : C:\Anaconda\envs
        package cache : C:\Anaconda\pkgs
         channel URLs : https://repo.continuum.io/pkgs/free/win-64/
                        https://repo.continuum.io/pkgs/free/noarch/
                        https://repo.continuum.io/pkgs/pro/win-64/
                        https://repo.continuum.io/pkgs/pro/noarch/
          config file : None
    is foreign system : False


C:\Users\lijin>

一个很棒的功能是 `conda` 可以产生一个自定义的环境，假设在安装的是**python 2.7**的情况下，想使用**python 3.4**，只需要在命令行下使用 `conda` 产生一个新的环境：

    conda create -n py34 python=3.4

这里这个环境被命名为 `py34` ，可以根据喜好将 `py34` 改成其他的名字。

使用这个环境时，只需要命令行下输入：

``` python
activate py34 #(windows)
source activate py34 #(linux, mac)
```

此时，我们的python版本便是 **`python 3.4`**了。

## spyder 编辑器

`Anaconda`使用的编辑器是`spyder`，可以在命令行下输入：

    spyder

来进入这个编辑器，具体使用方法不做介绍。